server/routerlicious/packages/services-shared/src/storage.ts

```diff
@@ -208,6 +209,9 @@ export class DocumentStorage implements IDocumentStorage {
 			throw error;
 		}
 
+		// Storage is known to take too long sometimes. Check timeout before continuing.
```

Comment:
👀 very curious about this, it sounds like a potential cause for the very sporadic test timeouts we see when running against t9s (running them locally, I managed to hit the timeout on the 149th run), and maybe also some tricker-to-interpret errors when running against r11s. Do we have any idea why it happens? cc @jikim-msft

Reply:
In AFR we use Azure Storage and I think that is what @znewton referred to here, I think. I was under the impression that t9s will run everything locally (as opposed to a production service like AFR) and therefore this comment might not apply.

Reply:
Got it. Yeah, in t9s everything is local. The only "remote" thing that happens is an http request to its own port when alfred calls historian (instead of a socket or just a call inside the process).

Reply:
I was also referring to the recent problem we've had with random service freeze-ups. For example, if Historian hangs for 15 minutes, that would trigger this error.

Reply:
Also, for context @alexvy86, we are using #17355 to set Node.js HTTP timeouts to 5 min to start, then will likely reduce it to 30s eventually to align with the default Chrome browser timeout.

Reply:
I'm definitely going to look into deploying a server version with that change to our internal cluster to see if we notice differences. Thanks for pointing it out!